sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -600,7 +600,8 @@ object FunctionRegistry {
       } else {
         // Otherwise, find a constructor method that matches the number of arguments, and use that.
         val params = Seq.fill(expressions.size)(classOf[Expression])
-        val f = constructors.find(_.getParameterTypes.toSeq == params).getOrElse {
+        val f = constructors.find(e => e.getParameterTypes.toSeq == params
+            || e.getParameterTypes.head == classOf[String]).getOrElse {
```

Comment:
Seems like it's less hacky to create a new `expressionWithAlias` method, with only the necessary logic
```
def expressionWithAlias ... = {
  val constructors = tag.runtimeClass.getConstructors.filter(c => e.getParameterTypes.head == classOf[String])
  assert(constructors.length == 1)
  try {
    constructors.head.newInstance(name, expressions : _*).asInstanceOf[Expression]
  } ...
}
```

Reply:
then we don't even need the `MultiNamedExpression` trait. We just need to register `bool_and`, `bool_or` with `expressionWithAlias`

Reply:
@cloud-fan updated as per your suggestions.